test/Microsoft.ML.Tests/AnomalyDetectionTests.cs

```diff
@@ -24,39 +18,52 @@ public AnomalyDetectionTests(ITestOutputHelper output) : base(output)
         }
 
         /// <summary>
-        /// RandomizedPcaTrainer test 
+        /// RandomizedPcaTrainer test.
         /// </summary>
         [Fact]
         public void RandomizedPcaTrainerBaselineTest()
         {
-            var mlContext = new MLContext(seed: 1, conc: 1);
-            string featureColumn = "NumericFeatures";
+            var trainPath = GetDataPath(TestDatasets.mnistOneClass.trainFilename);
+            var testPath = GetDataPath(TestDatasets.mnistOneClass.testFilename);
+            
+            var transformedData = DetectAnomalyInMnistOneClass(trainPath, testPath);
 
-            var reader = new TextLoader(Env, new TextLoader.Options()
-            {
-                HasHeader = true,
-                Separator = "\t",
-                Columns = new[]
-                {
-                    new TextLoader.Column("Label", DataKind.R4, 0),
-                    new TextLoader.Column(featureColumn, DataKind.R4, new [] { new TextLoader.Range(1, 784) })
-                },
-                AllowSparse = true
-            });
+            // Evaluate
+            var metrics = ML.AnomalyDetection.Evaluate(transformedData, k: 5);
+
+            Assert.Equal(0.98667, metrics.Auc, 5);
+            Assert.Equal(0.90000, metrics.DrAtK, 5);
+        }
+
+        /// <summary>
+        /// Test anomaly detection when the test data has no anomalies.
+        /// </summary>
+        [Fact]
+        public void NoAnomalyTest()
+        {
+            var trainPath = GetDataPath(TestDatasets.mnistOneClass.trainFilename);
 
-            var trainData = reader.Read(GetDataPath(TestDatasets.mnistOneClass.trainFilename));
-            var testData = reader.Read(GetDataPath(TestDatasets.mnistOneClass.testFilename));
+            var transformedData = DetectAnomalyInMnistOneClass(trainPath, trainPath);
 
-            var pipeline = ML.AnomalyDetection.Trainers.RandomizedPca(featureColumn);
+            Assert.Throws<ArgumentOutOfRangeException>(() => ML.AnomalyDetection.Evaluate(transformedData));
+        }
 
-            var transformer = pipeline.Fit(trainData);
-            var transformedData = transformer.Transform(testData);
+        private IDataView DetectAnomalyInMnistOneClass(string trainPath, string testPath)
+        {
+            var reader = ML.Data.CreateTextLoader(new[]
```

Comment:
>new[] [](start = 50, length = 5)

ensure to format the text ctr+k+d

Reply:
done, no changes suggested by vs2017 and vscode